usaspending_api/common/tests/unit/test_fiscal_year_helpers.py

```diff
@@ -69,41 +72,60 @@ def test_dates_are_fiscal_year_bookends():
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 8, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 9, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 10, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 11, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 12, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 1, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 2, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 3, 1)) == 4
-
-    # Test a fiscal year WAY in the future.  We should definitely have flying cars by then.
-    assert fyh.calculate_last_completed_fiscal_quarter(2010, date(2001, 3, 1)) is None
-
-    # And one test using the default as_of_date just to make sure nothing catches on fire.
-    assert fyh.calculate_last_completed_fiscal_quarter(2000) == 4
+    now = datetime.now(timezone.utc)
+    yesterday = now + timedelta(days=-1)
+    tomorrow = now + timedelta(days=1)
+    current_fy = fyh.generate_fiscal_year(now)
+
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2000,
+        submission_reveal_date=now,
+        submission_fiscal_quarter=1,
+        is_quarter=True,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2000,
+        submission_reveal_date=now,
+        submission_fiscal_quarter=2,
+        is_quarter=False,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2010,
+        submission_reveal_date=tomorrow,
+        submission_fiscal_quarter=2,
+        is_quarter=True,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=yesterday,
+        submission_fiscal_quarter=3,
+        is_quarter=True,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=now,
+        submission_fiscal_quarter=4,
+        is_quarter=True,
+    )
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=current_fy,
+        submission_reveal_date=tomorrow,
+        submission_fiscal_quarter=5,
+        is_quarter=True,
+    )
+
+    assert fyh.calculate_last_completed_fiscal_quarter(2000) == 1 # not 2, since is_quarter=False
+    assert fyh.calculate_last_completed_fiscal_quarter(2001) is None # no row in table for 2001
+    assert fyh.calculate_last_completed_fiscal_quarter(2010) is None # not revealed yet
+    assert fyh.calculate_last_completed_fiscal_quarter(current_fy) == 4 # not 3, since both are revealed & quarters; 5 not revealed yet
```

Comment:
> 5 not revealed yet

Didn't notice before that this test case had a fiscal_quarter set to 5. I think you can remove the bit of test data that has `submission_fiscal_quarter=5` since that will never occur realistically.

And that last assertion can be updated to just remove everything after the semicolon because it is still a good test.